drivers: lpadc: Move ADC clocking to SOC level

decsny opened the pull request:
To be consistent with the current NXP clocking scheme, move the LPADC clocking code to the SOC files where all of the other peripheral clocking is done.

Commit:
drivers: lpadc: Move SOC code out of driver

To be consistent with the current NXP clocking scheme,
move the LPADC clocking code to the SOC files where
all of the other peripheral clocking is done.

Also remove any other SOC-specific code to the
respective SOC file and out of this driver.

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

DerekSnell reviewed (commented):
Thanks @decsny ,

Other than missing some cpoyright updates, I don't see any issue with these changes.  Thanks

decsny commented on drivers/adc/adc_mcux_lpadc.c:
```diff
@@ -41,8 +41,6 @@ LOG_MODULE_REGISTER(nxp_mcux_lpadc);
 struct mcux_lpadc_config {
 	ADC_Type *base;
-	uint32_t clock_div;
-	uint32_t clock_source;
```
NXP does not have a copyright in this file

decsny commented:
DNM waiting on #56495 to merge first

danieldegrasse commented on drivers/adc/adc_mcux_lpadc.c:
```diff
@@ -412,27 +411,10 @@ static int mcux_lpadc_init(const struct device *dev)
-
-	/* Disable VREF power down */
+
+#if defined(CONFIG_SOC_LPC55S36)
```
If we're going to remove the remaining SOC specific code, why include this? Wouldn't it be better to move this to the LPC55S36 SOC init?

decsny replied:
yes

mmahadevan108 reviewed (commented):
Should the commit message change to `Move SoC-specific code out of the driver file.` as this change is moving more than clocking code.

Commit:
drivers: lpadc: re-add power level code

Code for power level property was accidentally
removed by accident from driver in commit 9921c59f40

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>
